server/src/main/java/org/elasticsearch/search/aggregations/bucket/range/RangeAggregator.java

```diff
@@ -215,15 +226,186 @@ public boolean equals(Object obj) {
         }
     }
 
-    final ValuesSource.Numeric valuesSource;
-    final DocValueFormat format;
-    final Range[] ranges;
-    final boolean keyed;
-    final InternalRange.Factory rangeFactory;
+    /**
+     * Build an {@link Aggregator} for a {@code range} aggregation. If the
+     * {@code ranges} can be converted into filters then it builds a
+     * {@link FiltersAggregator} and uses that to collect the results
+     * <strong>if</strong> that aggregator can run in "filter by filter"
+     * collection mode. If it can't then we'll collect the ranges using
+     * a native {@link RangeAggregator} which is significantly faster
+     * than the "compatible" collection mechanism for the filters agg.
+     */
+    public static Aggregator build(
+        String name,
+        AggregatorFactories factories,
+        ValuesSourceConfig valuesSourceConfig,
+        InternalRange.Factory<?, ?> rangeFactory,
+        Range[] ranges,
+        boolean keyed,
+        SearchContext context,
+        Aggregator parent,
+        CardinalityUpperBound cardinality,
+        Map<String, Object> metadata
+    ) throws IOException {
+        Aggregator adapted = adaptIntoFiltersOrNull(
+            name,
+            factories,
+            valuesSourceConfig,
+            rangeFactory,
+            ranges,
+            keyed,
+            context,
+            parent,
+            cardinality,
+            metadata
+        );
+        if (adapted != null) {
+            return adapted;
+        }
+        return buildWithoutAttemptedToAdaptToFilters(
+            name,
+            factories,
+            (ValuesSource.Numeric) valuesSourceConfig.getValuesSource(),
+            valuesSourceConfig.format(),
+            rangeFactory,
+            ranges,
+            keyed,
+            context,
+            parent,
+            cardinality,
+            metadata
+        );
+    }
 
-    final double[] maxTo;
+    public static Aggregator adaptIntoFiltersOrNull(
+        String name,
+        AggregatorFactories factories,
+        ValuesSourceConfig valuesSourceConfig,
+        InternalRange.Factory<?, ?> rangeFactory,
+        Range[] ranges,
+        boolean keyed,
+        SearchContext context,
+        Aggregator parent,
+        CardinalityUpperBound cardinality,
+        Map<String, Object> metadata
+    ) throws IOException {
+        if (valuesSourceConfig.fieldType() == null) {
+            return null;
+        }
+        if (false == valuesSourceConfig.fieldType().isSearchable()) {
+            return null;
+        }
+        if (valuesSourceConfig.missing() != null) {
+            return null;
+        }
+        if (valuesSourceConfig.script() != null) {
+            return null;
+        }
```

Comment:
Yes.

Reply:
Do you plan to do that in this PR, or open a ticket for it?  Could be a good candidate for `good first issue`, since it's mostly just moving code around.

Reply:
Actually, I see now that you added the method on VSConfig, but aren't calling it from here.  Oversight?